core/src/main/java/org/mskcc/cbio/portal/scripts/ImportMutationalSignatureData.java

```diff
@@ -0,0 +1,196 @@
+            boolean replace = commandLine.hasOption("replace");
+
+            ProgressMonitor.setConsoleMode(true);
+
+            File mutationalSignatureFile;
+            int numLines;
+            if(commandLine.hasOption("mut-sig-file")) {
+                mutationalSignatureFile = new File((String) commandLine.getOptionValue("mut-sig-file"));
+
+                System.out.println("Reading gene data from:  " + mutationalSignatureFile.getAbsolutePath());
+                numLines = FileUtil.getNumLines(mutationalSignatureFile);
+                System.out.println(" --> total number of lines:  " + numLines);
+                ProgressMonitor.setMaxValue(numLines);
+                ImportMutationalSignatureData.importData(mutationalSignatureFile, replace);
+            }
+            
+        } 
+        catch(ParseException ex){
+                System.out.println(ex.getMessage()); 
+        }
+        catch (RuntimeException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+    
+    /**
+     * Adds the mutational signatures parsed from the file into the Database.
+     * This is for importing mutational signature metadata
+     * @param mutationalSignatureFile File with mutational signature information
+     * @throws IOException
+     * @throws DaoException
+     */
+    
+    public static void importData (File mutationalSignatureFile, boolean replace) throws IOException, DaoException{
+        try{
+            FileReader reader = new FileReader(mutationalSignatureFile);
+            BufferedReader buf = new BufferedReader(reader);
+            String line;
+            while ((line = buf.readLine()) != null){
+                ProgressMonitor.incrementCurValue();
+                ConsoleUtil.showProgress();
+                
+                
+                String[] parts = line.split("\t");
+                boolean arrayIsEmpty = false;
+                for(int i = 0; i < parts.length; i++){
+                    if (parts[i].equals("") || parts[i] == null){
+                        arrayIsEmpty = true;
+                    }
+                    else{
+                        arrayIsEmpty = false;
+                    }
+                }
+
+                if (!arrayIsEmpty){
+                    //assume mutational signature id and description are first two columns in file
+                    MutationalSignature mutationalSignature = new MutationalSignature();
+                    mutationalSignature.setMutationalSignatureId(parts[0]);
+                    mutationalSignature.setDescription(parts[1]);
+
+                    //check if mutational signature already exists by the mutational signature id
+                    MutationalSignature existingMutationalSignature = DaoMutationalSignature.getMutationalSignatureById(mutationalSignature.getMutationalSignatureId());
+
+                    // only replaces description
+                    if (replace) {
```

Comment:
This looks backwards - you should check if the mutational signature exists and if it does, update only if replace is true.

if (existingMutationalSignature)
{
if (replace) replace;
}
// signature does not exist, add it
else {
add to database
}